From 4519fe0db5eb00e8a29b3eeb144e4ada4fe971dc Mon Sep 17 00:00:00 2001
From: Yixun Lan <dlan@gentoo.org>
Date: Fri, 20 Mar 2015 13:02:35 -0400
Subject: [PATCH 01/10] systemd: Allow specifying unit dir to configure

systemd isn't necessarily in the buildroot at the time that plymouth
gets built, so autodetection of the unitdir from pkgconfig isn't
necessarily feasible.

This commit adds a new option to configure, --with-systemdunitdir,
that lets distros specify the unit directory manually.

https://bugs.gentoo.org/show_bug.cgi?id=543712
---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 110cc69..445034d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -122,7 +122,12 @@ AM_CONDITIONAL(ENABLE_SYSTEMD_INTEGRATION, [test "$enable_systemd_integration" =
 
 if test x$enable_systemd_integration = xyes; then
   AC_DEFINE(PLY_ENABLE_SYSTEMD_INTEGRATION, 1, [Coordinate boot up with systemd])
-  SYSTEMD_UNIT_DIR=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)
+  AC_ARG_WITH([systemdunitdir], AC_HELP_STRING([--with-systemdunitdir=DIR],
+    [path to systemd service directory]), [path_systemdunit=${withval}],
+      [path_systemdunit="`$PKG_CONFIG --variable=systemdsystemunitdir systemd`"])
+  if (test -n "${path_systemdunit}"); then
+    SYSTEMD_UNIT_DIR="${path_systemdunit}"
+  fi
   AC_SUBST(SYSTEMD_UNIT_DIR)
 fi
 
-- 
1.9.0

